### ISSUE TITLE
Upgrade to videojs 5

### ISSUE THREAD

**Issue by doublex** [brightcove/videojs-thumbnails#30]
Are there any plans to upgrade this great plugin to videojs 5?

**Comment by xbgmsharp** [comment]
Yes, I will be done at some point.
Did you test? Is it not working?

**Comment by tindzk** [comment]
It works fine for me.

**Comment by Tavousi** [comment]
its not working on 5.4.4 error / Cannot read property 'progressControl' of undefined.

**Comment by tahola** [comment]
Yes, not working on 5.4.4, the seconds doenst match.

**Comment by gkatsev** [comment]
Yeah, this needs to get updated to videojs. Anyone want to take a crack at it? See https://github.com/videojs/video.js/wiki/5.0-Change-Details for the changes.

**Comment by tahola** [comment]
For anyonye interested, changing this line  
> mouseTime = Math.floor((left - progressControl.el().offsetLeft) / progressControl.width() * duration); 

by
> mouseTime = Math.floor(event.offsetX / progressControl.width() * duration); 

worked for me.

**Comment by Tavousi** [comment]
On 5.4.5 says :+1: Uncaught TypeError: Cannot read property 'progressControl' of undefined
<pre>
progressControl = player.controlBar.progressControl;
</pre>

**Comment by aequasi** [comment]
Same issue that @Tavousi has, here

**Comment by xbgmsharp** [comment]
Works for me using.
http://docs.brightcove.com/en/video-cloud/brightcove-player/guides/thumbnails-plugin.html

**Comment by bolt088** [comment]
For Video.js 5, in `videojs.thumbnails.js` I replaced this line:
>mouseTime = Math.floor((left - progressControl.el().offsetLeft) / progressControl.width() * duration);

with this

```
var hoverTime  = $(".vjs-mouse-display").attr("data-current-time").split(":");
var mouseTime = 0;
for(var i = b.length-1; i >= 0; i--)
    mouseTime += parseInt((i == hoverTime.length-1) ? (hoverTime[i]) : ((i == hoverTime.length-2) ? (hoverTime[i] * 60) : ((i == hoverTime.length-3) ? (hoverTime[i] * 60 * 60) : (0))));
```

And it is working perfectly in any way now.

**Comment by richardbushell** [comment]
Same issue as @Tavousi above...
Uncaught TypeError: Cannot read property 'progressControl' of undefined:-
progressControl = player.controlBar.progressControl;
Using vjs 5.16.0
Is there a known VJS Thumbnails plugin that works with vjs5? Preferably webvtt?

**Comment by ghost** [comment]
@iraniamir about the 

> Uncaught TypeError: Cannot read property 'progressControl' of undefined

in my case me the error triggers if I init the player and add the plugin within the same statement
```
var player = videojs("video", {
    plugins: {
        thumbnails: {
            // plugin settings here
        }
 });
```
but doesn't if I separate the player initialization from the plugin's one
```
player = videojs("myVideojsId");

player.thumbnails({
    // plugin settings
});
```

I'm using videojs 5.13.2 minified version.

**Comment by bernardoadc** [comment]
Thanks @mrcointreau , the same worked here, using `player.options.children`

**Comment by trisys3** [comment]
By now, you would need to be compatible with VideoJS 6-7, as well. To be clear, I'm not sure if adding compatibility with 5 won't automatically add compatibility with 6-7, just putting it out there.
